rclcpp/resource/logging.hpp.em

```diff
@@ -92,14 +99,25 @@ def is_supported_feature_combination(feature_combination):
  * \param ... The format string, followed by the variable arguments for the format string.
  * It also accepts a single argument of type std::string.
  */
-#define RCLCPP_@(severity)@(suffix)(logger, @(''.join([p + ', ' for p in get_macro_parameters(feature_combination).keys()]))...) \
+@{params = get_macro_parameters(feature_combination).keys()}@
+#define RCLCPP_@(severity)@(suffix)(logger, @(''.join([p + ', ' for p in params]))...) \
   do { \
     static_assert( \
       ::std::is_same<typename std::remove_cv<typename std::remove_reference<decltype(logger)>::type>::type, \
       typename ::rclcpp::Logger>::value, \
       "First argument to logging macros must be an rclcpp::Logger"); \
+@[ if 'throttle' in feature_combination]@ \
+    auto get_time_point = [&clock](rcutils_time_point_value_t * time_point) -> rcutils_ret_t { \
+    try { \
+      *time_point = clock.now().nanoseconds(); \
+    } catch (...) { \
+      return RCUTILS_RET_ERROR; \
```

Comment:
We should try to print something here, otherwise it silently fails. There is a "safe" write to stderr macro you could use, e.g.:

https://github.com/ros2/rcutils/blob/f8c58a19386a9cf31044a716d7e0946ad75766e8/src/split.c#L115-L117